example/lib/main.dart

```diff
@@ -19,41 +20,67 @@ class _MyAppState extends State<MyApp> {
   @override
   void initState() {
     super.initState();
-    initPlugin();
+    // Can't show a dialog in initState, delaying initialization
+    SchedulerBinding.instance.addPostFrameCallback((_) => initPlugin());
   }
 
   // Platform messages are asynchronous, so we initialize in an async method.
   Future<void> initPlugin() async {
     TrackingStatus status;
-    // Platform messages may fail, so we use a try/catch PlatformException.
-    try {
-      status = await AppTrackingTransparency.requestTrackingAuthorization();
-    } on PlatformException {
-      _authStatus = 'Failed to open tracking auth dialog.';
+
+    if (await AppTrackingTransparency.canRequestTrackingAuthorization()) {
+      // Show a custom explainer dialog before the system dialog
+      if (await showCustomTrackingDialog(context)) {
+        // Wait for dialog popping animation
+        await Future.delayed(const Duration(milliseconds: 200));
+        // Platform messages may fail, so we use a try/catch PlatformException.
+        try {
+          status = await AppTrackingTransparency.requestTrackingAuthorization();
+        } on PlatformException {
+          _authStatus = 'Failed to open tracking auth dialog.';
+        }
+        setState(() {
+          _authStatus = "$status";
+        });
+      }
     }
 
     final uuid = await AppTrackingTransparency.getAdvertisingIdentifier();
     print("UUID: $uuid");
-
-    if (!mounted) {
-      return;
-    }
-
-    setState(() {
-      _authStatus = "$status";
-    });
   }
 
+  Future<bool> showCustomTrackingDialog(BuildContext context) =>
```

Comment:
instead of the whole Cuppertino dialog stuff, it will be much cleaner to use regular material dialogues. Material approach seems defacto standart in flutter community.

Reply:
I find it better to keep the custom explainer dialog and iOS request tracking dialog in the same theme. Other widgets may be in material theme.

Reply:
Tracking is a ios specific feature, and you are right, it may be more appropriate to use the cuppertino themed widgets. But most of the time people are going to use a custom themed dialog box which matches their application style and colors. Is it our responsibility to recommend a style in such an example?

Reply:
Ok, you convinced me. I will implement the Material dialog.